### PR TITLE
Thumbnail upload 2.0

### PR DESCRIPTION
**Pull request by officialpranav**
Upload images to maximo item master. Note: The image name must be the same as the 9-series number of the item you are uploading to. If a thumbnail exists for the item you are uploading to, the tool will ~~_delete the existing image and upload the new one_~~ not upload the new image.

### CODE REVIEW

File: assets/common.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ class WorkerHandler {
                 log.error(e.data[1]);
             } else if (e.data[0] === 'update'){
                 updateItemStatus(e.data[1],e.data[2]);
+            } else if (e.data[0] === 'callback'){
+                callback(e.data.slice(1,));
             } else {
```
Comment (jonathanmajh) [inline assets/common.js:45]: callback is too generic please rename

File: assets/maximo.js [hunk 2]
```diff
@@ -242,17 +243,83 @@ class Maximo {
...
             headers: {
                 "filetype":"XML",
                 "apikey": this.login.userid,
+                //"preview": "1"
             },
             body: xmldoc,
         });
         let content = await response.json();
+        console.log(content);
         return parseInt(content.validdoc);
```
Comment (jonathanmajh) [inline assets/maximo.js:257]: remove log

File: assets/maximo.js [hunk 3]
```diff
@@ -242,17 +243,83 @@ class Maximo {
...
+                "apikey": this.login.userid,
+            }
+        })
+        content = await response.json();
+
+        //delete image if it exists
+        if(content["_imagelibref"]){
+            //console.log("image exists");
+            //delete existing image
+            response = await fetch(`https://test.manage.test.iko.max-it-eam.com/maximo/api/os/mxitem/${itemId}?action=system:deleteimage`, {
```
Comment (jonathanmajh) [inline assets/maximo.js:297]: should just show a warning for that item instead, doubtful anyone would check for existing images before hand

File: renderer/item_main.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ const { clipboard, ipcRenderer, shell } = require('electron');
 const Database = require('../assets/indexDB');
 const Validate = require('../assets/validators');
 let itemsToUpload = [];
+let imgsToUpload = [];
 let colLoc = {
```
Comment (jonathanmajh) [inline renderer/item_main.js:7]: really?

**Comment by jonathanmajh** [comment]
image upload fail should be more descriptive messages
image

**Review by jonathanmajh — CHANGES_REQUESTED** [review]

File: assets/maximo.js [hunk 5]
```diff
@@ -242,17 +243,83 @@ class Maximo {
...
         let content = await response.json();
+        console.log(content);
         return parseInt(content.validdoc);
     }
+
+    async uploadImageToMaximo(image){
+        //check valid image type
+        if(image.type !== "image/jpeg"){
+            return 'fail';
+        }
```
[the remark this answers is not in the document]
Comment (officialpranav) [reply assets/maximo.js:264]: when u try uploading in maximo it says only jpeg

File: renderer/item_main.js (hunk 4 above)
Comment (officialpranav) [reply renderer/item_main.js:7]: yes

File: assets/maximo.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jonathanmajh) [reply assets/maximo.js:264]: png definitely works though

**Review by jonathanmajh — CHANGES_REQUESTED** [review]

**Pull request merged**